packages/io/amf-deserializer/translate.js

```diff
@@ -2,93 +2,86 @@ const createObject = require('./objectBuilder')
 const parse = require('./parse')
 
 const translate = function (src, filename, options) {
-  options && options.statusCallback && options.statusCallback({progress: 0})
-  filename = filename || 'amf'
-  const defaults = {pxPmm: require('./constants').pxPmm, version: '0.0.0', addMetaData: true}
+  const defaults = {
+    pxPmm: require('./constants').pxPmm
+  }
   options = Object.assign({}, defaults, options)
-  const {version, pxPmm, addMetaData} = options
+  const { version, pxPmm, addMetaData } = options
+
+  options && options.statusCallback && options.statusCallback({ progress: 0 })
 
   // parse the AMF source
-  const {amfObj, amfMaterials, amfTextures, amfConstels} = parse(src, pxPmm)
+  const { amfObj, amfMaterials, amfTextures, amfConstels } = parse(src, pxPmm)
+
   // convert the internal objects to JSCAD code
   let code = addMetaData ? `//
-  // producer: OpenJSCAD.org ${version} AMF deserializer
-  // date: ${new Date()}
-  // source: ${filename}
-  //
-  ` : ''
+// producer: OpenJSCAD.org ${version} AMF deserializer
```

Comment:
ok, so this clarifies it : addMetadata is needed in defaults :)

Reply:
Both version and addMetadata are set by the defaults of deserialize().
I didn't include another set of defaults for instantiate() and translate().

Reply:
hmm I see your point, but it does mean that they cannot function independantly from deserialize anymore right ?
This is nagging me a bit: it means that technically we would need to have a check in place in the function to make sure there are no missing params (the reason for defaults) which is overkill?
I am ok with removing for now, but it seems somewhat wrong ? it might be a good idea to make this coherent across deserializers in the future

Reply:
Or document that those options (version and addMetaData) are REQUIRED
I could add that at least.

Reply:
sure !